### ISSUE TITLE
The ribbon panels do not match the FEM workbench 

### ISSUE THREAD

**Issue by raindropsfromsky** [APEbbers/FreeCAD-Ribbon#130]
The ribbon panels do not match the FEM workbench.

![image](https://github.com/user-attachments/assets/0aff5c75-ff0b-44da-bf0a-6b4201ecfd8c)
As a result, the ribbon for the FEM WB is not controlled.

Reporting for Ribbon WB 1.4.1 with FC 1.1.0dev on Windows 11

**Comment by hasecilu** [comment]
I got a message "wb FEM not loaded" from:

```python
        # ensure that workbench is already loaded
        workbench = Gui.activeWorkbench()
        if not hasattr(workbench, "__Workbench__"):
            # XXX for debugging purposes
            if Parameters_Ribbon.DEBUG_MODE is True:
                print(f"wb {workbench.MenuText} not loaded")

            # wait for 0.1s hoping that after that time the workbench is loaded
            timer.timeout.connect(self.onWbActivated)
            timer.setSingleShot(True)
            timer.start(100)
            return
```

Is the time for loading it small?

**Comment by APEbbers** [comment]
Hi,

in the develop branch, i've changed the timer to 500. Does that help?

**Comment by hasecilu** [comment]
No, I even change to much bigger values, maybe is something related to the external dependencies?

**Comment by APEbbers** [comment]
Does it help if you set the FEM workbench to autoload on startup?

**Comment by APEbbers** [comment]
In the developer branch i've changed the code:
![image](https://github.com/user-attachments/assets/39fee372-b2b1-4ae2-8c3a-70b329c2f820)

Does this help perhaps?

**Comment by hasecilu** [comment]
OK, `Develop` got problems but `main` is OK.

~~It seems not, also with latest changes now Ribbon is not loading~~

**ALL PREVIOUSLY NOT LOADED COMMADS NOW APPEAR**
Arch Linux, Qt6

<details>
  <summary><b>Click here to show/hide</b></summary>

```
FreeCAD Ribbon: https://github.com/APEbbers/FreeCAD-Ribbon
wb Part Design not loaded
Uncaught exception:
   File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 776, in onUserChangedWorkbench
    self.onWbActivated()
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name

AttributeError: 'PartDesignWorkbench' object has no attribute '__Workbench__'
Traceback (most recent call last):
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 776, in onUserChangedWorkbench
    self.onWbActivated()
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name
AttributeError: 'PartDesignWorkbench' object has no attribute '__Workbench__'
CRITICAL:pyqtribbon_local.logger:Uncaught exception:
   File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 776, in onUserChangedWorkbench
    self.onWbActivated()
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name

AttributeError: 'PartDesignWorkbench' object has no attribute '__Workbench__'
Traceback (most recent call last):
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 776, in onUserChangedWorkbench
    self.onWbActivated()
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name
AttributeError: 'PartDesignWorkbench' object has no attribute '__Workbench__'
wb Part Design not loaded
Uncaught exception:
   File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 1493, in __init__
    ribbon = ModernMenu()
             ^^^^^^^^^^^^
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 253, in __init__
    self.onUserChangedWorkbench()
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 776, in onUserChangedWorkbench
    self.onWbActivated()
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name

AttributeError: 'PartDesignWorkbench' object has no attribute '__Workbench__'
Traceback (most recent call last):
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 1493, in __init__
    ribbon = ModernMenu()
             ^^^^^^^^^^^^
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 253, in __init__
    self.onUserChangedWorkbench()
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 776, in onUserChangedWorkbench
    self.onWbActivated()
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name
AttributeError: 'PartDesignWorkbench' object has no attribute '__Workbench__'
CRITICAL:pyqtribbon_local.logger:Uncaught exception:
   File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 1493, in __init__
    ribbon = ModernMenu()
             ^^^^^^^^^^^^
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 253, in __init__
    self.onUserChangedWorkbench()
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 776, in onUserChangedWorkbench
    self.onWbActivated()
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name

AttributeError: 'PartDesignWorkbench' object has no attribute '__Workbench__'
Traceback (most recent call last):
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 1493, in __init__
    ribbon = ModernMenu()
             ^^^^^^^^^^^^
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 253, in __init__
    self.onUserChangedWorkbench()
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 776, in onUserChangedWorkbench
    self.onWbActivated()
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name
AttributeError: 'PartDesignWorkbench' object has no attribute '__Workbench__'
Init: Showing main window
Main window restored
Active view is Inicio (at 0x613365691370)
Show main window
Toolbars restored
<Dock> OverlayWidgets.cpp(812): restore OverlayLeft 0
<Dock> OverlayWidgets.cpp(812): restore OverlayRight 472
<Dock> OverlayWidgets.cpp(812): restore OverlayTop 
<Dock> OverlayWidgets.cpp(812): restore OverlayBottom 
Couldn't connect to spacenav daemon. Please ignore if you don't have a spacemouse.
Loading GUI of Spreadsheet module... done
wb Spreadsheet not loaded
Uncaught exception:
   File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name

AttributeError: 'SpreadsheetWorkbench' object has no attribute '__Workbench__'
Traceback (most recent call last):
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name
AttributeError: 'SpreadsheetWorkbench' object has no attribute '__Workbench__'
CRITICAL:pyqtribbon_local.logger:Uncaught exception:
   File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name

AttributeError: 'SpreadsheetWorkbench' object has no attribute '__Workbench__'
Traceback (most recent call last):
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name
AttributeError: 'SpreadsheetWorkbench' object has no attribute '__Workbench__'
3e-08 MainWindow.cpp(1721): update actions
Loading Fem module... done
Loading GUI of Fem module... done
wb FEM not loaded
Uncaught exception:
   File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name

AttributeError: 'FemWorkbench' object has no attribute '__Workbench__'
Traceback (most recent call last):
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name
AttributeError: 'FemWorkbench' object has no attribute '__Workbench__'
CRITICAL:pyqtribbon_local.logger:Uncaught exception:
   File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name

AttributeError: 'FemWorkbench' object has no attribute '__Workbench__'
Traceback (most recent call last):
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name
AttributeError: 'FemWorkbench' object has no attribute '__Workbench__'
4.89072 MainWindow.cpp(1721): update actions
Banco de trabajo FreeGrid inicializado v2.1.1
You are using'FreeGridWorkbench', 1
wb Sketcher not loaded
Uncaught exception:
   File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name

AttributeError: 'SketcherWorkbench' object has no attribute '__Workbench__'
Traceback (most recent call last):
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name
AttributeError: 'SketcherWorkbench' object has no attribute '__Workbench__'
CRITICAL:pyqtribbon_local.logger:Uncaught exception:
   File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name

AttributeError: 'SketcherWorkbench' object has no attribute '__Workbench__'
Traceback (most recent call last):
  File "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon/./FCBinding.py", line 797, in onWbActivated
    Gui.activateWorkbench(workbench.name())
                          ^^^^^^^^^^^^^^^^
  File "<string>", line 96, in name
AttributeError: 'SketcherWorkbench' object has no attribute '__Workbench__'
8.47128 MainWindow.cpp(1721): update actions
'PartDesignWorkbench', 1
```
</details>

**Comment by hasecilu** [comment]
The recent changes that fixed #77 seems to have affected on the FEM "Loads and Constraints custom", as mentioned this was fixed sometime ago but there is a regression on `Develop`

![image](https://github.com/user-attachments/assets/c28448a0-a61b-4737-a811-63c8f9a86350)

**Comment by APEbbers** [comment]
> The recent changes that fixed #77 seems to have affected on the FEM "Loads and Constraints custom", as mentioned this was fixed sometime ago but there is a regression on `Develop`
> 
> ![image](https://private-user-images.githubusercontent.com/53124818/392062264-c28448a0-a61b-4737-a811-63c8f9a86350.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMyNjMwMjAsIm5iZiI6MTczMzI2MjcyMCwicGF0aCI6Ii81MzEyNDgxOC8zOTIwNjIyNjQtYzI4NDQ4YTAtYTYxYi00NzM3LWE4MTEtNjNjOGY5YTg2MzUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAzVDIxNTIwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAwZDEzOGZmODNiNWI4MWM4OWQyNTZkNjFlODZiNjIwNjJmNzNkMzkzY2M5M2VmZmY3MTNjNTQ5YTMxNzI1NTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JYSCU-NqpN9kiYItO8mvtx39JSVIj8c9PaTwPll0Kk0)

This should now be solved again.
